src/Dapr.Actors/Runtime/ActorStateManager.cs

```diff
@@ -405,9 +551,11 @@ private StateMetadata(object value, Type type, StateChangeKind changeKind)
 
             public Type Type { get; }
 
-            public static StateMetadata Create<T>(T value, StateChangeKind changeKind)
+            public DateTimeOffset? TTLExpireTime { get; set; }
+
+            public static StateMetadata Create<T>(T value, StateChangeKind changeKind, DateTimeOffset? ttlExpireTime = null, TimeSpan? ttl = null)
```

Comment:
Is the intent to have a method which allows a TTL, expiry time, or neither?  If so, consider three methods, one which doesn't accept either, one that accepts the offset, and a third that accepts the time span.  That would help ensure proper usage.

Reply:
Yes, that makes sense- they are intended to be mutually exclusive. I've moved to using 3 functions.